Assets/Scripts/Framework/Statemachine/StateMachine.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class StateMachine : MonoBehaviour
+{
+    [Header("Base StateMachine")]
+    [SerializeField] protected BaseState startingState;
+    [SerializeField] protected BaseState[] states;
+    [SerializeField] protected BaseState currentState;
+
+    [SerializeField] private float removeStateQueueTime;
+    
+    private readonly List<BaseState> _switchStateQueue = new List<BaseState>();
+
+    protected void Awake()
+    {
+        SetStatesParent();
+    }
+
+    private void Start()
+    {
+        EnterStartingState();
+    }
+
+    protected void Update()
+    {
+        currentState.UpdateState(this);
+    }
+    protected void FixedUpdate()
+    {
+        currentState.FixedUpdateState(this);
+
+        if (!currentState.IsValidToSwitch || _switchStateQueue.Count <= 0) return;
+        
+        SwitchState(_switchStateQueue[0]);
+        _switchStateQueue.Remove(_switchStateQueue[0]);
+    }
+
+    private void SetStatesParent()
+    {
+        foreach (var state in states)
+        {
+            state.SetParent(this);
+        }
+    }
+
+    private void EnterStartingState()
+    {
+        currentState = startingState;
+        currentState.EnterState(this);
+    }
+
+    /// <summary>
+    /// Switch the current targetState to a different one.
+    /// Is it valid to switch targetState?
+    /// </summary>
+    /// <param name="targetState">Give target state to switch into.</param>
+    /// <param name="isCalledFromExitState">If the SwitchState is called form a ExitState function, this needs to be true!</param>
+    public void SwitchState(BaseState targetState, bool isCalledFromExitState = false)
```

Comment:
Why is this the only method so far with a summary? Why specifically this one?

Reply:
Omdat de parameter misschien confussing kan zijn. Had tijdens het gebruik wat error's dus moest het op een manier fixen.

Reply:
Wil je het weg dan?

Reply:
No it's fine, I was just wondering why this one and not other methods!